lib/ecto/changeset/relation.ex

```diff
@@ -29,6 +29,7 @@ defmodule Ecto.Changeset.Relation do
   """
   def empty?(%{cardinality: _}, %NotLoaded{}), do: true
   def empty?(%{cardinality: :many}, []), do: true
+  def empty?(%{cardinality: :many}, changes), do: Enum.all?(changes, & &1.action == :replace)
```

Comment:
Should we do: `& &1.action in [:replace, :delete]`? Since `:delete` can also be set manually in the changeset function?